### PR TITLE
Hashing UIDs with salt

### PR DESCRIPTION
**Pull request by axg3073**

### CODE REVIEW

File: server/src/resolvers/usersResolver.ts [hunk 1]
```diff
@@ -52,11 +56,18 @@ const UsersResolvers = {
       },
       context: any
     ) => {
-      const hashedUniversityID = hashUniversityID(args.universityID);
+
+      const length = 16; //salt length
+
+      const salt = randomBytes(Math.ceil(length/2))
```
Comment (MatthewGalan) [inline server/src/resolvers/usersResolver.ts:62]: Why are we doing `Math.ceil(length/2)`? Why not just put do `randomBytes(16)`?

File: server/src/resolvers/usersResolver.ts [hunk 2]
```diff
@@ -52,11 +56,18 @@ const UsersResolvers = {
...
+        .toString('hex')
+        .slice(0,length);
```
Comment (MatthewGalan) [inline server/src/resolvers/usersResolver.ts:64]: Why are we calling `slice`?

File: server/src/resolvers/roomsResolver.ts [hunk 3]
```diff
@@ -63,8 +63,18 @@ const RoomResolvers = {
       const room = await RoomRepo.getRoomByID(args.roomID);
       assert(room);
 
-      const hashedUniversityID = hashUniversityID(args.universityID);
-      const user = await UserRepo.getUserByUniversityID(hashedUniversityID);
+      let hashedUIDString = "";
+
+      const users = await UserRepo.getUsers();
+      users.forEach(user => {
```
Comment (MatthewGalan) [inline server/src/resolvers/roomsResolver.ts:69]: Why are we looping through every user in the system? Can't we just grab the hashed UID from the current user using the callback argument in `ifAllowed`?

File: server/src/resolvers/usersResolver.ts [hunk 4]
```diff
@@ -3,16 +3,20 @@ import * as ModuleRepo from "../repositories/Training/ModuleRepository";
...
 }
 
-export function hashUniversityID(universityID: string) {
-  return createHash("sha256").update(universityID).digest("hex");
+export function hashUniversityID(universityID: string, salt: string) {
+  const hashed = createHash("sha256").update(universityID + salt).digest("hex");
+  return {
+    salt: salt,
+    hashedValue: hashed
+  };
```
Comment (MatthewGalan) [inline server/src/resolvers/usersResolver.ts:19]: What's the benefit of returning the hash and salt separately, as an object? I think if we just returned `hashed + ":" + salt`, then we wouldn't have to stitch them together every time we want to check a UID.

**Review by MatthewGalan — CHANGES_REQUESTED** [review]

File: server/src/resolvers/roomsResolver.ts (hunk 3 above)
Comment (axg3073) [reply server/src/resolvers/roomsResolver.ts:69]: will the current User always be the swiped in user? If so, would makes sense with that route

**Pull request closed without merging**